### PR TITLE
Add E0019 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #24407.
cc @michaelsproul

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -279,6 +279,41 @@ println!("{}", Y);
 ```
 "##,
 
+E0019: r##"
+A function call isn't allowed in the const's initialization expression
+because the expression's value must be known at compile-time. Bad example:
```
Comment (pnkfelix) [inline src/librustc/diagnostics.rs]: The phrase "Bad example" is often used when one is saying that the example itself is bad -- i.e. when the example does not actually serve as a demonstration of the topic at hand.

I would instead write: "Example of erroneous code:" or "Example of non-compiling code:"

Comment (GuillaumeGomez) [inline src/librustc/diagnostics.rs]: I actually had the same comment on the other PRs I opened yesterday. I forgot to update this one, I do it.

**Comment by GuillaumeGomez** [comment]
Updated!

File: src/librustc/diagnostics.rs [hunk 2]
```diff
@@ -279,6 +279,41 @@ println!("{}", Y);
...
+}
+
+fn main() {
+    const FOO: Test = Test::V1;
+
+    const A: i32 = FOO.test(); // You can't call Test::func() here !
+}
+```
+
+Remember: you can't use a function call inside a const's initialization
```
Comment (pnkfelix) [inline src/librustc/diagnostics.rs]: You should add an additional paragraph about the `const fn` feature, which is currently feature-gated and thus only available in Nightly (but there are other error explanations like E0152 that reference feature-gated things).  The sole purpose of `const fn` is *to be* a function call that breaks the rule you wrote here.  :)

Here is a demonstration of `const fn`:

  https://play.rust-lang.org/?gist=449fb78958b96eca7528&version=nightly

**Comment by pnkfelix** [comment]
(I've been discussing the question of whether to include discussion of feature-gated things.  For now I think it would be best if @GuillaumeGomez removes the final commit above from this PR, and then we land it.  We can then put that final commit up as a separate PR that will only be landed after we resolve the question of how to discuss feature-gated things in error explanations.)

**Pull request closed without merging**